### PR TITLE
Add Slider package

### PR DESCRIPTION
**Pull request by LucaT1**
This PR adds the slider to this collection of Material Design components. 🚀 

The Slider has the following functionalities:

- Can be dragged anywhere

- On focusing, it can move by using the arrows on the keyboard

TODO: 📝 

- [x] Fix thumb from going outside from the slider

- [x] Add more keyboard events

- [x] Fix some movement issues

### CODE REVIEW

File: packages/Slider/src/index.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    this.moveSlider(e)
+  }
+
+  @bind
+  handleMouseUp(e) {
+    this.setState({ focus: false })
+  }
+
+  @bind
+  handleMouseMove(e) {
```
Comment (Gejsi) [inline packages/Slider/src/index.js]: The **handleMouseMove** event should be applied to the **document** itself when the **handleMouseDown** event is fired so the user will have more freedom to move the slider.

File: packages/Slider/src/index.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+      // Enter: Focs/unfocus
+      case 13:
+        this.setState({ keyFocus: !keyFocus })
+      break
+
+      // Forward: increment of 1
+      case 39:
+        if(keyFocus && progress != 100)
+          this.setState({ progress: progress+1 })
```
Comment (Gejsi) [inline packages/Slider/src/index.js]: The **increment** should happen even when pressing the **arrow_key_up**, simply add another case.

File: packages/Slider/src/index.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      break
+
+      // Backwards: decrease of 1
+      case 37:
+        if(keyFocus && progress != 0)
+          this.setState({ progress: progress-1 })
```
Comment (Gejsi) [inline packages/Slider/src/index.js]: - The **decremen**t should happen even when pressing the **arrow_key_down**, simply add another case.

- Use the vise function to make sure that the user can't go out from the slider.

File: packages/Slider/src/index.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      break
+      default:
+      break
+    }
+  }
+
+  @bind
+  getStyles() {
+    const { progress, keyFocus, focus } = this.state
+
```
Comment (Gejsi) [inline packages/Slider/src/index.js]: Add **background** prop to make the slider themeable

**Comment by LucaT1** [comment]
Fixed merge conflicts. Ready to merge! :tada:

File: packages/Slider/src/indicator.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      color: color || 'rgba(255, 255, 255, .87)',
+      borderRadius: '48.8%',
+      textTransform: 'uppercase',
+      fontSize: 12,
+      transition: 'transform .3s cubic-bezier(0.4, 0.0, 0.2, 1)',
+      userSelect: 'none',
+      height: 32,
+      width: 32,
+      transformOrigin: 'bottom',
+      transform: `translateX(-50%) ${focus ? 'scale(1, 1)' : 'scale(0, 0)'}`,
```
Comment (LucaT1) [inline packages/Slider/src/indicator.js]: Add a space after this line

File: packages/Slider/src/indicator.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        borderRight: '12px solid transparent',
+        borderTop: `12px solid ${background || '#2196F3'}`,
+      }
+    }
+
+    return styles
+  }
+
+  render({ value, max }) {
+    const styles = this.getStyles.bind(this)()
```
Comment (LucaT1) [inline packages/Slider/src/indicator.js]: Use decko's `@bind` instead of the ugly `.bind(this)()`

File: packages/Slider/src/indicator.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+    return(
+      <div
+        style={styles}
+        onMouseEnter={this.handleMouseEnter}
```
Comment (LucaT1) [inline packages/Slider/src/indicator.js]: Useless event listener

File: packages/Slider/src/indicator.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        onMouseLeave={this.handleMouseLeave}>
```
Comment (LucaT1) [inline packages/Slider/src/indicator.js]: Useless event listener

**Comment by LucaT1** [comment]
LGTM 👌

**Pull request merged**